src/main/kotlin/lotto/domain/Buyer.kt

```diff
@@ -9,11 +10,17 @@ class Buyer {
     private val tickets: MutableList<LotteryTicket> = mutableListOf()
 
     fun buyLotteryTickets(
-        payedPrice: Int,
+        buyingData: BuyingData,
         lotteryTicketMaker: LotteryTicketMaker = DefaultLotteryTicketMaker()
     ): List<LotteryTicket> {
-        repeat(payedPrice / lotteryTicketMaker.lotteryTicketPrice) {
-            tickets.add(lotteryTicketMaker.createLotteryTicket())
+        var ticketCount = buyingData.inputPrice / lotteryTicketMaker.lotteryTicketPrice
+
+        for (lottoNumbers in buyingData.manualNumbersList) {
+            tickets.add(lotteryTicketMaker.createManualLotteryTicket(lottoNumbers))
+        }
+
+        repeat(ticketCount - buyingData.manualTicketCount) {
+            tickets.add(lotteryTicketMaker.createAutoLotteryTicket())
         }
         return tickets
```

Comment:
외부에서 컬렉션을 수정할 수 있기 때문에 Mutable 컬렉션을 드러낼 때에는 방어적 복사를 해주세요.
List 타입으로 반환하더라도 구현체는 MutableList 이기 때문에 악의적으로 캐스팅을 해서 데이터를 변경할 수 있습니다.

```suggestion
        return tickets.toList()
```